fix: decrement memberWithdrawableBalance and safetyNetBalance on withdrawal execution

franrolotti opened the pull request:
Closes #38

This PR ensures that both `memberWithdrawableBalance` and `safetyNetBalance` are decremented when a withdrawal request is executed (small, auto-executed, or consensus-approved).

franrolotti committed:
fix: decrement memberWithdrawableBalance and safetyNetbalance on large withdrawal execution

franrolotti committed:
test: expect DuplicateMember revert when members array contains duplicates

bagelface commented on test/unit/SafetyNetUnit.sol:
```diff
@@ -267,14 +267,15 @@ contract SafetyNetUnit is Test {
   }
 
-  function test_CreateWhenMembersArrayContainsDuplicates() external {
+  function test_CreateWhenMembersArrayContainsDuplicates_reverts() external {
```
don't need to add "_reverts". None of the other tests that test revert have this in test name

franrolotti replied:
c12ab4e

bagelface commented on src/contracts/SafetyNet.sol:
```diff
@@ -256,6 +263,13 @@ contract SafetyNet is ISafetyNet, ReentrancyGuard, OwnableUpgradeable {
 
     if (_request.yesVotes > _safetyNet.members.length * _safetyNet.consensusThreshold / 100) {
       // Consensus reached - execute withdrawal immediately
+      if (memberWithdrawableBalance[_request.safetyNetId][_request.owner] < _request.amount) {
```
same code twice. looks like a good candidate for a private function

franrolotti replied:
c12ab4e

bagelface reviewed (changes requested):
a few comments

franrolotti committed:
refactor: deduplicate withdrawal checks via _deduct; keep test naming consistent

franrolotti commented:
@bagelface comments addressed

RonTuretzky commented:
@franrolotti i see that ci lint is failling?

RonTuretzky commented on src/contracts/SafetyNet.sol:
```diff
@@ -75,6 +75,15 @@ contract SafetyNet is ISafetyNet, ReentrancyGuard, OwnableUpgradeable {
+    }
+    memberWithdrawableBalance[_safetyNetId][_member] -= _amount;
+    safetyNetBalance[_safetyNetId] -= _amount;
+  }
```
@franrolotti make sure that when you introduce new functions you are abiding by natspec conventions

franrolotti replied:
@RonTuretzky sorry about that, already fixed

RonTuretzky reviewed (changes requested):
I see a lot of linting issues, missing proper natspec documentatino on new function that was introduced

franrolotti committed:
fix: reorder _deduct; add NatSpec

franrolotti committed:
style: apply lint and formatting fixes across contracts, scripts, and tests

franrolotti commented:
@RonTuretzky  

I addressed the changes around the new function:

* Added proper NatSpec to the _deduct helper.
* Fixed function ordering so it complies with solhint’s ordering rule.
* Ensured large withdrawals deduct from both memberWithdrawableBalance and safetyNetBalance via _deduct.

The remaining lint findings you see in CI are from unrelated areas (e.g., max-states-count on SafetyNet.sol and private-vars-leading-underscore in fuzz tests) and from formatting differences between local stable Foundry and CI’s nightly formatter.

Do you want me to tackle these repo-wide lint/format issues in this PR, or should we keep this PR focused and follow up once the CI/tooling configuration is finalized?

franrolotti committed:
style: apply nightly formatter and update package.json for coverage

bagelface commented:
> @franrolotti i see that ci lint is failling?

CI has been failing for this repo for a while, it was just disabled. I recently reenabled it and made this ticket https://github.com/BreadchainCoop/safety-net/issues/51
